examples/mixed-spmv/mixed-spmv.cpp

```diff
@@ -0,0 +1,271 @@
+SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+******************************<GINKGO LICENSE>*******************************/
+
+// @sect3{Include files}
+
+// This is the main ginkgo header file.
+#include <ginkgo/ginkgo.hpp>
+
+// Add the fstream header to read from data from files.
+#include <fstream>
+// Add the C++ iostream header to output information to the console.
+#include <iostream>
+// Add the STL map header for the executor selection.
+#include <map>
+// Add the string manipulation header to handle strings.
+#include <string>
+// Add the timing header for timing.
+#include <chrono>
+// Add the random header to generate random vectors.
+#include <random>
+
+namespace {
+
+
+/**
+ * Generate a random value.
+ *
+ * @tparam ValueType  valuetype of the value
+ * @tparam ValueDistribution  type of value distribution
+ * @tparam Engine  type of random engine
+ *
+ * @param value_dist  distribution of array values
+ * @param engine  a random engine
+ *
+ * @return ValueType
+ */
+template <typename ValueType, typename ValueDistribution, typename Engine>
+typename std::enable_if<!gko::is_complex_s<ValueType>::value, ValueType>::type
+get_rand_value(ValueDistribution &&value_dist, Engine &&gen)
+{
+    return value_dist(gen);
+}
+
+/**
+ * Specialization for complex types.
+ *
+ * @copydoc get_rand_value
+ */
+template <typename ValueType, typename ValueDistribution, typename Engine>
+typename std::enable_if<gko::is_complex_s<ValueType>::value, ValueType>::type
+get_rand_value(ValueDistribution &&value_dist, Engine &&gen)
+{
+    return ValueType(value_dist(gen), value_dist(gen));
+}
+
+/**
+ * timing the apply operation A->apply(b, x). It will runs 2 warmup and get
+ * average time among 10 times.
+ *
+ * @return seconds
+ */
+double timing(std::shared_ptr<const gko::Executor> exec,
+              std::shared_ptr<gko::LinOp> A, std::shared_ptr<gko::LinOp> b,
+              std::shared_ptr<gko::LinOp> x)
+{
+    int warmup = 2;
+    int rep = 10;
+    for (int i = 0; i < warmup; i++) {
+        A->apply(lend(b), lend(x));
+    }
+    double total_sec = 0;
+    for (int i = 0; i < rep; i++) {
+        auto xx = x->clone();
+        exec->synchronize();
+        auto start = std::chrono::steady_clock::now();
+        A->apply(lend(b), lend(xx));
+        exec->synchronize();
+        auto stop = std::chrono::steady_clock::now();
+        std::chrono::duration<double> duration_time = stop - start;
+        total_sec += duration_time.count();
+        if (i + 1 == rep) {
+            // copy the result back to x
+            x->copy_from(lend(xx));
+        }
```

Comment:
Could you add a few comments here, especially the synchronization might be surprising?